NAudio/Midi/MetaEvent.cs

```diff
@@ -44,6 +44,11 @@ public MetaEvent(MetaEventType metaEventType, int metaDataLength, long absoluteT
         }
 
         /// <summary>
+        /// Creates a deep clone of this MIDI event.
+        /// </summary>
+        public override MidiEvent Clone() => new MetaEvent(metaEvent, metaDataLength, AbsoluteTime) { data = (byte[])data?.Clone() };

```

Comment:
Have had a look at the code and it would make sense for there to be an `UnknownMetaEvent` (or perhaps `CustomMetaEvent` or `NonStandardMetaEvent`) class. Feel free to introduce one and add that to this PR, as it won't be a breaking change since old code can still create them as `MetaEvent`